enterprise/models.py

```diff
@@ -1571,41 +1578,55 @@ def contains_programs(self, program_uuids):
 
         return bool(program_uuids and set(program_uuids).issubset(content_ids_in_catalog))
 
-    def get_course(self, course_key):
+    def get_course(self, course_key, request=None):
```

Comment:
A small suggestion might be to use `crum` to get the current request, if it exists to avoid needing to pass `request` through to `get_course`, `get_course_run`, `get_course_and_course_run `, and `get_program`. 

```python
request = crum.get_current_request()
```

Reply:
Good tip, I didn't know about that!